Sources/Sentry/Public/SentryEvent.h

```diff
@@ -196,4 +196,24 @@ NS_SWIFT_NAME(Event)
 
 @end
 
+/**
+ * Subclass of SentryEvent so we can add the Decodable implementation via a Swift extension. We need
+ * this due to our mixed use of public Swift and ObjC classes. We could avoid this class by
+ * converting SentryReplayEvent back to ObjC, but we rather accept this tradeoff as we want to
+ * convert all public classes to Swift in the future. This class needs to be public as we can't add
+ * the Decodable extension implementation to a class that is not public.
+ *
+ * @note: We can’t add the extension for Decodable directly on SentryEvent, because we get an error
+ * in SentryReplayEvent: 'required' initializer 'init(from:)' must be provided by subclass of
+ * 'Event' Once we add the initializer with required convenience public init(from decoder: any
+ * Decoder) throws { fatalError("init(from:) has not been implemented")
+ * }
+ * we get the error initializer 'init(from:)' is declared in extension of 'Event' and cannot be
+ * overridden. Therefore, we add the Decodable implementation not on the Event, but to a subclass of
+ * the event.
+ */
+@interface SentryEventDecodable : SentryEvent
```

Comment:
Please let me know your thoughts about this somewhat ugly but practical workaround. I'm open to suggestions on how we can avoid this. As already pointed out, I don't want to convert the SentryReplayEvent back to ObjC, but maybe it's a better approach than this one.

Reply:
I agree with not converting back to Objective-C.

If this approach works, It seems fine to me.